lib/nrf_modem_lib/Kconfig.modemlib

```diff
@@ -220,10 +220,10 @@ config NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC
 	  Let the application define the fault handler function.
 
 config NRF_MODEM_LIB_ON_FAULT_LTE_NET_IF
-	bool "Application defined"
+	bool "Pass to network interface driver"
```

Comment:
Well, in the end the network interface driver will emit a `NET_EVENT_CONN_IF_FATAL_ERROR` error, which must be handled by the application. That's not what "pass to network interface driver" suggests, as in, the network interface driver is handling the fault.

What about "Raise a fatal connectivity error network event" ? Bit long, but it reflects the description of the event

![image](https://github.com/nrfconnect/sdk-nrf/assets/1151289/a09ba872-dd2d-4c32-a871-7012fb05bc67)

and does not suggest that the event is handled by the network interface driver (it isn't).